3 of 3: Documentation cleanup and update. Added a motif finding tutorial.

rjurney opened the pull request:
<!--
Thanks for sending a pull request!  Here are some tips for you:
  1. Ensure you have added or run the appropriate tests for your PR
  2. If the PR is unfinished, add '[WIP]' in your PR title, e.g., '[WIP]Your PR title ...'.
  3. Be sure to keep the PR description updated to reflect all changes.
  4. Please write your PR title to summarize what this PR proposes.
  5. If possible, provide a concise example to reproduce the issue for a faster review.
-->
This is a documentation only change, the docs part of #473. The docs build and look okay when I serve them.

### What changes were proposed in this pull request?
<!--
Please clarify what changes you are proposing. The purpose of this section is to outline the changes and how this PR fixes the issue. 
If possible, please consider writing useful notes for better and faster reviews in your PR. See the examples below.
  1. If you refactor some codes with changing classes, showing the class hierarchy will help reviewers.
  2. If there is design documentation, please add the link.
  3. If there is a discussion in the mailing list, please add the link.
-->

I added a network motif tutorial and updated / cleaned the docs generally. I added the new google group and the new logo to the images folder.

### Why are the changes needed?
<!--
Please clarify why the changes are needed. For instance,
  1. If you propose a new API, clarify the use case for a new API.
  2. If you fix a bug, you can clarify why it is a bug.
-->

The docs were wrong in a few ways and there weren't even instructions in the README, which stopped new users from onboarding.

rjurney commented:
This project should be merged following #512 and #513, with #513.

SemyonSinchenko commented on README.md:
```diff
@@ -1,38 +1,209 @@
 # graphframes
```
Let's add a logo here?

rjurney replied:
Done! https://github.com/graphframes/graphframes/blob/rjurney/motif-tutorial-docs/README.md

rjurney commented:
Okay @bjornjorgensen @SauronShepherd @SemyonSinchenko I am done here. I would maybe suggest reviewing it by reading [docs/README.md](docs/README.md) and running `jekyll` to view the documentation. Give it a once over. I went through and fixed anything that was old, broken, out of date version, etc. I added new community resources and the `graphframes-py` package I reserved that will go out with 0.8.5. It's a big PR but this is english not code, and it cross-refers to itself and depends on the changes in the other two PRs this is based on: #512 <-- #518 <-- #511

rjurney commented:
@SemyonSinchenko this one is hard to review... lots of text changes. I have double checked it, but appreciate any review you can give :)

SemyonSinchenko commented on README.md:
```diff
@@ -1,39 +1,219 @@
-Users can write highly expressive queries by leveraging the DataFrame API, combined with a new
-API for motif finding.  The user also benefits from DataFrame performance optimizations
-within the Spark SQL engine.
+This is a package for DataFrame-based graphs on top of Apache Spark. Users can write highly expressive queries by leveraging the DataFrame API, combined with a new API for network motif finding. The user also benefits from DataFrame performance optimizations within the Spark SQL engine. GraphFrames works in Java, Scala, and Python.
```
```suggestion
This is a package for graphs processing and analytics on scale. It is build on top of Apache Spark and relying on DataFrame abstraction. Users can write highly expressive queries by leveraging the DataFrame API, combined with a new API for network motif finding. The user also benefits from DataFrame performance optimizations within the Spark SQL engine. GraphFrames works in Java, Scala, and Python.
```

SemyonSinchenko commented on README.md:
```diff
@@ -1,39 +1,219 @@
+
+## GraphFrames is Back!
+
+This projects was dormant for some time, but we are happy to announce that it is now back in active development! We are working on a new release with many bug fixes and improvements. We are also working on a new website and documentation.
```
```suggestion
This projects was in maintenance mode for some time, but we are happy to announce that it is now back in active development! We are working on a new release with many bug fixes and improvements. We are also working on a new website and documentation.
```

SemyonSinchenko commented on README.md:
```diff
@@ -1,39 +1,219 @@
-API for motif finding.  The user also benefits from DataFrame performance optimizations
-within the Spark SQL engine.
+This is a package for DataFrame-based graphs on top of Apache Spark. Users can write highly expressive queries by leveraging the DataFrame API, combined with a new API for network motif finding. The user also benefits from DataFrame performance optimizations within the Spark SQL engine. GraphFrames works in Java, Scala, and Python.
+
```
Let's add a note about deprecation of GraphX in Spark 4.0? Something like "With deprecation of GraphX in Apache Spark Core GraphFrames is the recommended approach for the distributed graphs processing."?

SemyonSinchenko commented on docs/img/4-node-directed-graphlets.png:
Where is this one taken? I can assume that it is from a scientific paper that may have a license like Creative Commons or a proprietary one. I think it is important to maintain IP clearance if we are going to move GraphFrames to the Spark core one day. Could you please add a source or even better to add a `NOTICE.md` file to the root of project with mentioning of the source of that picture and how is it licensed?

rjurney replied:
@SemyonSinchenko It's from this paper, which the caption and image itself lead to if you click: https://www.computer.org/csdl/journal/tb/2017/06/07501518/13rRUxcsYKf

The thing is... there is no open license image of graphlets. What if I include it now and seek permission in the meanwhile? Then I can figure out some alternative and in the meanwhile people can still learn about graphlets?

rjurney replied:
I reached out to two of the authors, asking for permission. It would be best to secure their permission. In reading this guide to fair use, I believe this image may qualify: https://www.copyrightlaws.com/simple-guide-fair-use/

rjurney replied:
I got permission to use the image :)

SemyonSinchenko commented on docs/img/Directed-Graphlet-G17.png:
The same

SemyonSinchenko commented on docs/img/Directed-Graphlet-G22.png:
The same

SemyonSinchenko commented on docs/img/G11_motif.png:
The same

SemyonSinchenko commented on docs/img/G4_and_G5_directed_network_motif.png:
The same

SemyonSinchenko reviewed (commented):
LGTM overall, thank you! The only important comment is about licenses. Other comments are minor.
